lib/chef/client.rb

```diff
@@ -451,9 +473,9 @@ def sync_cookbooks
     #
     # @api private
     def setup_run_context
-      run_context = policy_builder.setup_run_context(specific_recipes)
+      @run_context = policy_builder.setup_run_context(specific_recipes, run_context)
       assert_cookbook_path_not_empty(run_context)
-      run_status.run_context = run_context
+      run_status.run_context = run_context # backcompat for chefspec
```

Comment:
should this be @run_context since you just assigned to that?

Reply:
no, its an accessor so it works fine and returns `@run_context`